pkg/querier/queryrange/query_range.go

```diff
@@ -187,7 +188,7 @@ func (prometheusCodec) MergeResponse(responses ...Response) (Response, error) {
 	return &response, nil
 }
 
-func (prometheusCodec) DecodeRequest(_ context.Context, r *http.Request) (Request, error) {
+func (prometheusCodec) DecodeRequest(_ context.Context, r *http.Request, headers []string) (Request, error) {
```

Comment:
`forwardHeaders`, which is more explanatory and consist with the command-line flag.